graylog2-web-interface/src/views/logic/searchtypes/messages/MessageSortConifg.js

```diff
@@ -0,0 +1,36 @@
+// @flow strict
+import Direction from 'views/logic/aggregationbuilder/Direction';
+import SortConfig, { Builder } from 'views/logic/aggregationbuilder/SortConfig';
+
+export type MessageSortConifgJson = {
+  field: string,
+  order: 'ASC' | 'DESC',
+};
+
+export default class MessageSortConifg {
```

Comment:
@dennisoelkers we talked about the usage of an extra class to transform the message sort configs when calling the api. I've tried to create a class which is as simple as possible, but it would be great to hear your opinion on how to define the relation with the `SortConfig` class. It feels a bit wrong to use e.g. the `SortConfig` Builder for the `fromJSON` method.

Reply:
I think we should extract a common base class that is used by both `MessageSortConfig` and `SortConfig`. Maybe it would make sense to rename `SortConfig` to `AggregateSortConfig` and use `SortConfig` as the name for the base class. Shared code would go in there and the two specialized classes would use it internally to hold their state, but customize construction & (de-)serialization.